catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java

```diff
@@ -0,0 +1,700 @@
+
+    private static final String DISABLED_CONFIGURATION_SUFFIX = "_disabled";
+
+    private static final String ID = "id";
+
+    private static final String SHORTNAME = "shortname";
+
+    private static final String CONFIGURATION_FILTER =
+            "(&(%s=%s)(|(service.factoryPid=*source*)(service.factoryPid=*Source*)(service.factoryPid=*service*)(service.factoryPid=*Service*)))";
+
+    private static final String CREATED_TOPIC = "ddf/catalog/event/CREATED";
+
+    private static final String UPDATED_TOPIC = "ddf/catalog/event/UPDATED";
+
+    private static final String DELETED_TOPIC = "ddf/catalog/event/DELETED";
+
+    private static final int SHUTDOWN_TIMEOUT_SECONDS = 60;
+
+    private ConfigurationAdmin configurationAdmin;
+
+    private FederationAdminService federationAdminService;
+
+    private MetaTypeService metaTypeService;
+
+    private Parser parser;
+
+    private ParserConfigurator unmarshalConfigurator;
+
+    private ExecutorService executor;
+
+    private String urlBindingName;
+
+    private Map<String, String> bindingTypeToFactoryPidMap = new HashMap<>();
+
+    private List<String> sourceActivationPriorityOrder = new ArrayList<>();
+
+    private Boolean activateConfigurations;
+
+    private Boolean preserveActiveConfigurations;
+
+    private Boolean cleanUpOnDelete;
+
+    private SlotTypeHelper slotHelper;
+
+    private RegistryPackageTypeHelper registryTypeHelper;
+
+    public SourceConfigurationHandler(FederationAdminService federationAdminService,
+            ExecutorService executor) {
+        this.federationAdminService = federationAdminService;
+        this.executor = executor;
+    }
+
+    public void destroy() {
+        executor.shutdown();
+        try {
+            if (!executor.awaitTermination(SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+                executor.shutdownNow();
+                if (!executor.awaitTermination(SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+                    LOGGER.error("Thread pool didn't terminate");
+                }
+            }
+        } catch (InterruptedException e) {
+            executor.shutdownNow();
+        }
+    }
+
+    @Override
+    public void handleEvent(Event event) {
+        Metacard mcard = (Metacard) event.getProperty(METACARD_PROPERTY);
+        if (mcard == null || !mcard.getTags()
+                .contains(RegistryConstants.REGISTRY_TAG)) {
+            return;
+        }
+
+        executor.execute(() -> {
+            if (event.getTopic()
+                    .equals(CREATED_TOPIC)) {
+                processCreate(mcard);
+            } else if (event.getTopic()
+                    .equals(UPDATED_TOPIC)) {
+                processUpdate(mcard);
+            } else if (event.getTopic()
+                    .equals(DELETED_TOPIC)) {
+                processDelete(mcard);
+            }
+        });
+
+    }
+
+    protected void processCreate(Metacard metacard) {
```

Comment:
Did we make those `protected` because we expect them to be overridden by subclasses? If so, we should provide some JavaDoc, otherwise we should consider making them `private`.